src/Object/Hydrator.php

```diff
@@ -141,6 +141,29 @@ public static function get(object $object, string $property): mixed
         return self::accessibleProperty($object, $property)->getValue($object);
     }
 
+    /**
+     * @template T of object
+     *
+     * @param T $object
+     * @param T $other
+     */
+    public static function hydrateFromOtherObject(object $object, object $other): void
+    {
+        $classes = [$object::class, ...\array_values(\class_parents($object))];
+
+        $properties = [];
+        foreach ($classes as $class) {
+            $reflector = new \ReflectionClass($class);
+            foreach ($reflector->getProperties() as $property) {
+                $properties[$property->getName()] = $property->getName();
+            }
+        }
+
+        foreach ($properties as $property) {
+            self::set($object, $property, self::get($other, $property), catchErrors: true);
+        }
+    }
```

Comment:
I hope this code won't break too much things...

actually if it breaks, it means that the entity/object has a design problem :sweat_smile: